modules/karpenter-provisioners/main.tf

```diff
@@ -0,0 +1,82 @@
+locals {
+  karpenter_provisioner_values = yamlencode({ provisioners = [for node in var.karpenter_node_pools :
+    {
+      name = node.name
+      spec = {
+        ttlSecondsAfterEmpty = node.ttl_seconds_after_empty
+        labels               = node.kubernetes_labels
+        annotations          = node.annotations
+
+        consolidation = {
+          enabled = node.ttl_seconds_after_empty == null ? true : false
+        }
+
+        requirements = node.requirements
+
+        providerRef = {
+          name = node.name
+        }
+
+        taints = node.taints
+
+        startupTaints = node.startup_taints
+
+        kubeletConfiguration = node.kubelet_configuration
+
+        limits = node.limits
+
+        ttlSecondsUntilExpired = node.ttl_seconds_until_expired
+
+        ttlSecondsAfterEmpty = node.ttl_seconds_after_empty
+
+        weight = node.weight
+      }
+    }
+    ]
+
+    nodetemplates = [for temp in var.node_templates :
+      {
+        name = temp.name
+        spec = {
+          amiFamily          = temp.ami_family
+          tags               = module.karpenter_provisioner_label[node.name].tags
+          detailedMonitoring = temp.detailed_monitoring
+          instanceProfile    = temp.instance_profile
+          userData           = temp.user_data
+
+          subnetSelector = temp.subnet_selector
+          #{
+          #  aws-ids = join(",", local.private_subnet_ids) # TODO
+          #}
+
+          securityGroupSelector = temp.security_group_selector
+          #{
+          #  aws-ids = join(",", [data.aws_eks_cluster.saas.vpc_config[0].cluster_security_group_id]) # TODO
+          #}
+
+          amiSelector = temp.ami_selector
+          #{
+          #  aws-ids = join(",", [for arch in node.arch : data.aws_ami.karpenter_provisioner[arch].id])
+          #}
+
+          metadataOptions = temp.metadata_options
+
+          blockDeviceMappings = temp.block_device_mappings
+        }
+      }
+    ]
+    }
+  )
+}
+
+module "karpenter_provisioner_label" {
```

Comment:
rather doing that you can copy [context](https://github.com/rallyware/terraform-argocd-aws-eks-cluster-bootstrap/blob/main/context.tf) to the this module and iterate over the context, for example:
```hcl
module "node_template_label" {
for_each = { for node in var.karpenter_node_pools : node.name => node }

 source  = "cloudposse/label/null"
 version = "0.25.0"

context = module.this.context
attributes = [each.value.name]
tags = merge(module.this.tags, each.value.kubernetes_labels)


}

Reply:
Updated